tracer/src/Datadog.Trace/Configuration/SettingsManager.cs

```diff
@@ -0,0 +1,220 @@
+﻿// <copyright file="SettingsManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Datadog.Trace.Configuration.ConfigurationSources;
+using Datadog.Trace.Configuration.ConfigurationSources.Telemetry;
+using Datadog.Trace.Configuration.Telemetry;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Configuration;
+
+public partial record TracerSettings
+{
+    internal class SettingsManager(
+        TracerSettings tracerSettings,
+        MutableSettings initialMutable,
+        ExporterSettings initialExporter)
+    {
+        private readonly TracerSettings _tracerSettings = tracerSettings;
+        private readonly List<SettingChangeSubscription> _subscribers = [];
+        private SettingChanges? _latest;
+
+        /// <summary>
+        /// Gets the initial <see cref="MutableSettings"/>. On app startup, these will be the values read from
+        /// static sources. To subscribe to updates to these settings, from code or remote config, call <see cref="SubscribeToChanges"/>.
+        /// </summary>
+        public MutableSettings InitialMutableSettings { get; } = initialMutable;
+
+        /// <summary>
+        /// Gets the initial <see cref="ExporterSettings"/>. On app startup, these will be the values read from
+        /// static sources. To subscribe to updates to these settings, from code or remote config, call <see cref="SubscribeToChanges"/>.
+        /// </summary>
+        public ExporterSettings InitialExporterSettings { get; } = initialExporter;
+
+        /// <summary>
+        /// Subscribe to changes in <see cref="MutableSettings"/> and/or <see cref="ExporterSettings"/>.
+        /// Called whenever these settings change. If the settings have already changed when <see cref="SubscribeToChanges"/>
+        /// is called, <paramref name="callback"/> is invoked immediately with the latest configuration.
+        /// Also note that calling <see cref="SubscribeToChanges"/> twice with the same callback
+        /// will invoke the callback twice.
```

Comment:
```suggestion
        /// Subscribe to changes in <see cref="MutableSettings"/> and/or <see cref="ExporterSettings"/>.
        /// <paramref name="callback"/> is called synchronously whenever these settings change. If the settings have already changed when <see cref="SubscribeToChanges"/>
        /// is called, <paramref name="callback"/> is invoked immediately with the latest configuration.
        /// Also note that calling <see cref="SubscribeToChanges"/> twice with the same callback
        /// will invoke the callback twice. Callbacks should complete quickly to avoid blocking other operations.
```

Reply:
Yeah, "called synchronously whenever these settings change" doesn't really make sense I think 😅  The callback _is_ called synchronously, but it's called synchronously from the "updating thread", so it's asynchronous from the PoV of the person calling `SubscribeToChanges()`.

Like you say, "the settings have already changed ... callback is invoked immediately" _is_ called synchronously, so I've tweaked the wording